relations/src/relations/update_account.rs

```diff
@@ -0,0 +1,95 @@
+use halo2_base::{
+    gates::{GateChip, GateInstructions},
+    poseidon::hasher::{spec::OptimizedPoseidonSpec, PoseidonHasher},
+    utils::BigPrimeField,
+    AssignedValue,
+};
+#[allow(unused_imports)]
+use halo2_base::{
+    Context,
+    QuantumCell::{Constant, Existing, Witness},
+};
+
+use crate::{
+    account::CircuitAccount,
+    poseidon_consts::{RATE, R_F, R_P, T_WIDTH},
+};
+
+pub struct UpdateAccountInput<F, A>
+where
+    F: BigPrimeField,
+    A: CircuitAccount<F>,
+{
+    //public inputs
+    pub old_account_hash: AssignedValue<F>,
+    pub new_account_hash: AssignedValue<F>,
+    pub operation: A::Op,
```

Comment:
not `A::Op::OpPub`?

Reply:
Update is defined on whole `Op`. At least in docs